typescript/packages/plugins/ionic/src/ionic.service.ts

```diff
@@ -0,0 +1,146 @@
+import { Tool } from "@goat-sdk/core";
+import { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { IonicSdk } from "@ionicprotocol/sdk";
+import { z } from "zod";
+import { parseUnits, formatUnits } from "viem";
+import type { HealthMetrics, SupplyPosition, BorrowPosition } from "./types";
+
+export class IonicService {
+  private sdk: IonicSdk;
+  private supportedTokens: string[];
+
+  constructor(chainId: number, supportedTokens?: string[]) {
+    this.supportedTokens = supportedTokens || ["USDC", "WETH", "MODE"];
+  }
+
+  private initSdk(walletClient: EVMWalletClient) {
+    if (!this.sdk) {
+      this.sdk = new IonicSdk(
+        walletClient.publicClient,
+        walletClient.walletClient,
```

Comment:
This does not work. These are not exposed in walletClient since other types of wallets like Crossmint Smart Wallets and so would not have a `walletClient` and `publicClient`